python/make_variable.h

```diff
@@ -24,8 +24,13 @@ template <class T> struct MakeVariable {
     py::buffer_info info = valuesT.request();
     Dimensions dims(labels, {info.shape.begin(), info.shape.end()});
     auto var = variances
-                   ? makeVariable<T>(Dimensions{dims}, Values{}, Variances{})
-                   : makeVariable<T>(Dimensions(dims));
+                   ? makeVariable<T>(
+                         Dimensions{dims},
+                         Values(dims.volume(), core::default_init_elements),
```

Comment:
How does the improvement here work? Prevent the need to reallocate after in-correct sizing for values variances?

Reply:
It is not about reallocation, but about writing to memory twice (first init with zeros, then copy values). You can think of it just like the difference between

```cpp
std::vector<double> vec(size);
for (...)
  vec[i] = other[i];
```
and
```cpp
std::vector<double> vec;
vec.reserve(size);
for (...)
  vec.push_back(other[i]);
```